src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSScoreUtils.java

```diff
@@ -0,0 +1,426 @@
+                    .flatMap(read -> getValidHits(read, minCoverage, minIdentity).stream()
+                            .map(contig -> refNameToTaxBroadcast.value().containsKey(contig) ? refNameToTaxBroadcast.value().get(contig) : null)
+                            .filter(Objects::nonNull).distinct()
+                    ).collect(Collectors.groupingBy(Function.identity())).values(); //Group the flattened stream by tax ID into lists
+
+            final int numReads = (int) Utils.stream(readIter).count();
+
+            //Count the length of each list - if it's equal to the number of reads, we have a hit
+            final List<String> hitTaxIds = taxIDLists.stream()
+                    .filter(ids -> ids.size() == numReads)
+                    .map(ids -> ids.iterator().next())
+                    .collect(Collectors.toList());
+
+            final PSPathogenAlignmentHit info = new PSPathogenAlignmentHit(hitTaxIds, numReads);
+
+            //If there was at least one hit, append a tag to each read with the list of hits
+            if (hitTaxIds.size() > 0) {
+                final StringBuilder stringBuilder = new StringBuilder();
+                for (final String hit : hitTaxIds) {
+                    stringBuilder.append(hit);
+                    stringBuilder.append(",");
+                }
+                final String hitString = stringBuilder.substring(0, stringBuilder.length() - 1);
+                Utils.stream(readIter).forEach(read -> read.setAttribute(HITS_TAG, hitString));
+            } else {
+                Utils.stream(readIter).forEach(GATKRead::setIsUnmapped);
+            }
+            return new Tuple2<>(readIter, info);
+        });
+    }
+
+
+    /**
+     * Gets set of sufficiently well-mapped hits
+     */
+    private static Collection<String> getValidHits(final GATKRead read, final double minCoverage, final double minIdentity) {
+
+        //Short circuit if read isn't mapped
+        if (read.isUnmapped()) return new ArrayList<>(0);
+
+        //Check that NM tag is set
+        if (!read.hasAttribute("NM")) {
+            throw new UserException.BadInput("SAM flag indicates a read is mapped, but the NM tag is absent");
+        }
+
+        //Find and return all alignments meeting the coverage/identity criteria
+        final Collection<String> hits = new ArrayList<>();
+        if (isValidAlignment(read.getCigar(), read.getAttributeAsInteger("NM"), minCoverage, minIdentity)) {
+            hits.add(SAMSequenceRecord.truncateSequenceName(read.getAssignedContig()));
+        }
+        hits.addAll(getValidAlternateHits(read, "XA", 0, 2, 3, minCoverage, minIdentity));
+        hits.addAll(getValidAlternateHits(read, "SA", 0, 3, 5, minCoverage, minIdentity));
+        return hits;
+    }
+
+    /**
+     * Parses alternate alignments tag of form TAG:Z:val_A0,val_A1,val_A2,...;val_B0,val_B1,val_B2,...;...
```

Comment:
Can you expand this comment to cover the fact that you expect different values in the tag (ie the CIGAR and numMismatches)